### ISSUE TITLE
ENOENT and ELIFECYCLE errors when running the browser example

### ISSUE THREAD

**Issue by psuzzi** [theia-ide/theia#254]
ENOENT and ELIFECYCLE errors when running the browser example under macOS

**Steps to reproduce:** 
- open the [Developing.md#tldr](https://github.com/theia-ide/theia/blob/master/doc/Developing.md#tldr) link, and copy the code under "To build and run the browser example:"
- execute the code, and you will see the error reported below: 

```
npm WARN read-shrinkwrap This version of npm is compatible with lockfileVersion@1, but npm-shrinkwrap.json was generated for lockfileVersion@0. I'll try to do my best with it!
npm ERR! path /Users/psuzzi/git/theia/examples/browser/theia-core-0.0.1-alpha.2.tgz
npm ERR! code ENOENT
npm ERR! errno -2
npm ERR! syscall stat
npm ERR! enoent ENOENT: no such file or directory, stat '/Users/psuzzi/git/theia/examples/browser/theia-core-0.0.1-alpha.2.tgz'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent 

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/psuzzi/.npm/_logs/2017-06-26T22_42_06_570Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 254
npm ERR! browser-example@0.0.1-alpha.1 bootstrap: `npm install ../../config/local-dependency-manager && npm run localinstall`
npm ERR! Exit status 254
npm ERR! 
npm ERR! Failed at the browser-example@0.0.1-alpha.1 bootstrap script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm WARN Local package.json exists, but node_modules missing, did you mean to install?
```

Attached, you can see the two log files mentioned in the error log

[2017-06-26T22_42_06_570Z-debug.log.txt](https://github.com/theia-ide/theia/files/1103698/2017-06-26T22_42_06_570Z-debug.log.txt)

[2017-06-26T22_42_06_606Z-debug.log.txt](https://github.com/theia-ide/theia/files/1103700/2017-06-26T22_42_06_606Z-debug.log.txt)

**System details:** 

macOS Sierra 10.12.5

```
$ node -v && npm -v && python --version
v8.1.2
5.0.4
Python 2.7.10
```

**Comment by akosyakov** [comment]
You should use npm4, we have not migrated to npm5 yet.

**Comment by psuzzi** [comment]
A similar error persists after downgrading to npm 4 (`npm install npm@4 -g`) and re-running the script. 

```
env: node\r: No such file or directory
npm ERR! file sh
npm ERR! code ELIFECYCLE
npm ERR! errno ENOENT
npm ERR! syscall spawn
npm ERR! browser-example@0.0.1-alpha.1 localinstall: `ldm install --dev --original-sources`
npm ERR! spawn ENOENT
npm ERR! 
npm ERR! Failed at the browser-example@0.0.1-alpha.1 localinstall script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/psuzzi/.npm/_logs/2017-06-27T06_40_53_637Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! browser-example@0.0.1-alpha.1 bootstrap: `npm install ../../config/local-dependency-manager && npm run localinstall`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the browser-example@0.0.1-alpha.1 bootstrap script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/psuzzi/.npm/_logs/2017-06-27T06_40_53_664Z-debug.log
```

See the two logfiles attached

[2017-06-27T06_40_53_637Z-debug.log.txt](https://github.com/theia-ide/theia/files/1104376/2017-06-27T06_40_53_637Z-debug.log.txt)
[2017-06-27T06_40_53_664Z-debug.log.txt](https://github.com/theia-ide/theia/files/1104377/2017-06-27T06_40_53_664Z-debug.log.txt)

**System details:**

```
$ node -v && npm -v && python --version
v8.1.2
4.6.1
Python 2.7.10
```

**Comment by akosyakov** [comment]
`node\r` looks weird

**Comment by psuzzi** [comment]
Additional info: 
I installed node with `brew install node`, as a result it installed node 8 with npm 5.
After that, I downgraded to npm 4 with  `npm install npm@4 -g`
As [npm 4 is installed with node 7](https://nodejs.org/en/download/releases/), my feeling is that node 8 won't work well with npm 4. 
Plus, it [seems that node 7 is not available with homebrew](https://github.com/Homebrew/homebrew-core/issues/14118)

**Comment by kittaakos** [comment]
@psuzzi, you should rather use `nvm`. Then `nvm install && nvm use 7` should work out of the box.

**Comment by kittaakos** [comment]
You can get `nvm` from [here](https://github.com/creationix/nvm#install-script).

**Comment by psuzzi** [comment]
I tried installing as you suggested, but I still get an error similar to the initial one: 

```
env: node\r: No such file or directory

npm ERR! Darwin 16.6.0
npm ERR! argv "/Users/psuzzi/.nvm/versions/node/v7.10.0/bin/node" "/Users/psuzzi/.nvm/versions/node/v7.10.0/bin/npm" "run" "localinstall"
npm ERR! node v7.10.0
npm ERR! npm  v4.2.0
npm ERR! file sh
npm ERR! code ELIFECYCLE
npm ERR! errno ENOENT
npm ERR! syscall spawn
npm ERR! browser-example@0.0.1-alpha.1 localinstall: `ldm install --dev --original-sources`
```
I searched about the error `env: node\r: No such file or directory` , and I found [this github issue](https://github.com/nwjs-community/nw-builder/issues/75), which I think it could help.

**Comment by akosyakov** [comment]
You should look into `config/local-dependency-manager/bin/ldm`. It looks like there is `\r` character and ldm is trying to run `node\r` instead of `node`.

**Comment by psuzzi** [comment]
`config/local-dependency-manager/bin/ldm` looked ok. So, my feeling was the error was caused by line ending encoding. Indeed, I verified with `git config -l` that `core.autocrlf` was set to `true`, while, according to the [github page "dealing with line ending"](https://help.github.com/articles/dealing-with-line-endings/), on mac, `core.autocrlf` should be set to `input`.

I _resolved by_ setting `git config --global core.autocrlf input` and re running the command to build and launch Theia. Now everything works fine.

I suggest updating the documentation specifying that, for mac, git should have  `core.autocrlf=input`.
